### PR TITLE
Add use cases to README

### PR DESCRIPTION
**Pull request by nmattia**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,152 @@ necessary for fetching and updating the packages.
...
+
+This covers three use cases:
+
+* [Bootstrapping a Nix project](#bootstrapping-a-nix-project).
+* [Tracking a different a different nixpkgs branch](#tracking-a-nixpkgs-branch).
+* [Fetching packages from custom URLs](#using-custom-urls).
+
+#### Bootstrapping a Nix project
+
+There is an `init` command that is useful when starting a new Nix project or
```
Comment (zimbatm) [inline README.md]:
```suggestion
`niv init` is useful when starting a new Nix project or
```

File: README.md [hunk 2]
```diff
@@ -39,6 +39,152 @@ necessary for fetching and updating the packages.
...
+```
+
+The files `default.nix`, `nix/default.nix` and `nix/packages.nix` are useful
+defaults for any Nix project. The file `nix/sources.json` is the file used by
+niv to store versions and is initialized with niv and nixpkgs:
+
+``` json
+{
+    "nixpkgs": {
+        "url": "https://github.com/NixOS/nixpkgs-channels/archive/....tar.gz",
```
Comment (zimbatm) [inline README.md]: it might be best to put some real data as an example

File: README.md [hunk 3]
```diff
@@ -39,6 +39,152 @@ necessary for fetching and updating the packages.
...
+        "owner": "NixOS",
+        "branch": "nixos-18.09",
+        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz",
+        "repo": "nixpkgs-channels",
+        "sha256": "...",
+        "description": "Nixpkgs/NixOS branches that track the Nixpkgs/NixOS channels",
+        "rev": "..."
+    },
+    "niv": {
+        "homepage": "",
```
Comment (zimbatm) [inline README.md]: why is there an empty homepage here?

File: README.md [hunk 4]
```diff
@@ -39,6 +39,152 @@ necessary for fetching and updating the packages.
...
+update to the latest commit:
+
+``` shell
+$ niv update nixpkgs
+```
+
+To change the branch being tracked run this command:
+
+``` shell
+$ niv update -b nixos-19.03             # equivalent to --branch nixos-19.03
```
Comment (zimbatm) [inline README.md]:
```suggestion
$ niv update nixpkgs -b nixos-19.03             # equivalent to --branch nixos-19.03
```

Comment (zimbatm) [reply README.md]: or is it defaulting to `nixpkgs`?

File: README.tpl.md [hunk 5]
```diff
@@ -39,38 +39,186 @@ necessary for fetching and updating the packages.
   `nix/sources.nix` file that returns the sources as a Nix object.
 * [Show](#show): shows the packages' information.
 
+The next two sections cover [common use cases](#use-cases) and [full command
```
Comment (zimbatm) [inline README.tpl.md]: what is the difference between the template and the normal readme?

Comment (nmattia) [reply README.tpl.md]: The normal readme is generated from the template. It inlines the output of `niv <cmd> --help`.

File: README.md (hunk 4 above)
Comment (nmattia) [reply README.md]: Typo, it doesn't default to anything.

File: README.md (hunk 3 above)
Comment (nmattia) [reply README.md]: Not sure, the GitHub API returning silly stuff

**Pull request merged**